syntax = "proto3";

package calculator;

// Define a gRPC service
service Calculator {
  rpc Add (AddRequest) returns (AddResponse);
  rpc Multiply (MultiplyRequest) returns (MultiplyResponse);
}

// Request and Response messages
message AddRequest {
  int32 num1 = 1;
  int32 num2 = 2;
}

message AddResponse {
  int32 result = 1;
}

message MultiplyRequest {
  int32 num1 = 1;
  int32 num2 = 2;
}

message MultiplyResponse {
  int32 result = 1;
}

